### DISCUSSION TITLE
running eclair on raspiblitz

### DISCUSSION THREAD

**Discussion by BitromNO** [ACINQ/eclair#2676] [Node operators]
Hi! 
Did anyone try to run eclair on a raspiblitz node?
or others like:
- myNode
- EmbassyOS (startOS)
- Umbrel
or is it easier to just do a install on debian?

**Comment by sstone** [comment]
Hi,
Eclair runs on the JVM (which should be available for most platforms) but also needs to load native bindings for secp256k1. We provide such bindings for `linux arm64` so there's a good chance that it will run on a raspberry pi without any changes though we have not tried ourselves (if it does not work out of the box you will have to build https://github.com/ACINQ/secp256k1-kmp yourself). 
Like many JVM apps, Eclair will probably not run very well on machine with very little RAM (not because its memory use is inefficient, but because the JVM itself is an app and requires some memory for its own use, as well as some for the apps it loads and runs).

**Reply by BitromNO** [reply]
ok. im running on a amd64 machine 8gb ram, internal SSD disks. would be nice to test it! 
is linux debian the go to linux for eclair also?

**Comment by sstone** [comment]
There will probably be issues on distros that either do not use glibc (like Alpine) or use a bleeding edge version of glibc (Arch, there you will have to compile JNI bindings for secp256k1 yourself as explained above) but  AFAIK Eclair works fine on most standard 64bits linux distributions (debian or red hat based, including AWS Linux which we use in production).

**Reply by BitromNO** [reply]
Ok, thanks! 
I would like to use my Proxmox setup for a test. 
Is there a step by step instruction for noobs? 

- Debian with ssh enabled 
- Bitcoin Core (or can i Connect to a local Core install over LAN?)

- Eclair

- Can my imac be a potential candidate with osx?

**Comment by sstone** [comment]
We don't really have a step by step guide but installing and running eclair should be fairly simple:

- install a Java 11 JVM (go to https://adoptium.net/ if you don't know where to find one)
- install Bitcoin Core, with the right parameters (see https://github.com/ACINQ/eclair#prerequisite-bitcoin-core), and wait for bitcoin core to finish syncing
- create a configuration file for eclair (see https://github.com/ACINQ/eclair/blob/master/README.md#configuration). There are many options, but to get started you just need to specify which chain you're on (mainnet or testnet) and how to access bitcoin core, enable the API and you're good to go

And please, please start on `testnet` first :)

You can connect to a bitcoin node on a different machine on your LAN, you just need to make sure that it can be accessed from the machine where Eclair is running (you can test this with `bitcoin-cli`). Please note that it is not recommended to have your Bitcoin Core's API reachable from the internet. 
I'm pretty sure that an imac will do but it seems to be a desktop machine. If you want to route payments your node will have to be online 24/7

**Comment by chrisguida** [comment]
I built eclair for start9 a while back: https://github.com/chrisguida/eclair-wrapper

works just fine in docker
